homeassistant/components/zwave_js/light.py

```diff
@@ -404,6 +405,30 @@ def _calculate_color_values(self) -> None:
             CommandClass.SWITCH_COLOR,
             value_property_key=ColorComponent.COLD_WHITE.value,
         )
+        return (red_val, green_val, blue_val, ww_val, cw_val)
+
+    @callback
+    def _calculate_color_support(self) -> None:
+        """Calculate light colors."""
+        (red_val, green_val, blue_val, ww_val, cw_val) = self._get_color_values()
+        # RGB support
+        if red_val and green_val and blue_val:
+            self._supports_color = True
+        # color temperature support
+        if ww_val and cw_val:
+            self._supports_color_temp = True
+        # only one white channel (warm white) = rgbw support
+        elif red_val and green_val and blue_val and ww_val:
+            self._supports_rgbw = True
+        # only one white channel (cool white) = rgbw support
+        elif cw_val:
+            self._supports_rgbw = True
```

Comment:
Are these two checks correct?
Why don't we need to check `red_val and green_val and blue_val` in the 2nd check?

(This is not new code, it's just moved in this PR)

Reply:
Good question. I've been thinking the same thing about this code. I don't know enough details of the spec or devices to answer it.

Reply:
I have some vague memory that it's impossible to have a `cw_val` without also having all the others, but we should make if clear in a comment if that's the case.